src/prototype_room_my.js

```diff
@@ -520,14 +520,15 @@ Room.prototype.executeRoom = function() {
 
 Room.prototype.reviveRoom = function() {
   let nextRoomers = _.filter(Game.creeps, c => c.memory.role === 'nextroomer' &&
-    c.memory.routing.targetRoom === this.name).length;
+    c.memory.routing.targetRoom === this.name);
+  let nextRoomersTicksAmount = _(nextRoomers).map('ticksToLive').sum() / CREEP_LIFE_TIME;
```

Comment:
comment in code ?
It use average nextroomers ticks for 'count' how many nextroomers there are. Because previously, if there are multiple nextroomers dying at the same time , spawn will need time for respawn them and the room will wait without nextroomers the time they spawn and the time they need to do the travel.
Whith this, it calculate the sum of nextroomers ticksleft / 1500 wich give 5 for exemple if the 5 nextroomers have 1500 ticks left but it will give 4 for exemple if a nextroomer is gonna die soon so the spawn will spawn it faster and it will come in room earlier.